Releases - DevOps for ASP.NET Core Developers eBook V1.1 (#23784)
dotnet/docs: docs/architecture/devops-for-aspnet-developers/monitoring.md

## Log streaming

App and web server logs can be streamed in real time through the portal.

1. Open the [Azure portal](https://portal.azure.com), and navigate to the *mywebapp\<unique_number\>* App Service.
1. In the menu to the left, scroll down to the **Monitoring** section and select **Log stream**.

    ![Screenshot showing log stream link](./media/monitoring/log-stream.png)

Logs can also be [streamed via Azure CLI or Azure PowerShell](/azure/app-service/web-sites-enable-diagnostic-log#streamlogs), including through the Cloud Shell.

## Alerts

Azure Monitor also provides [real time alerts](/azure/monitoring-and-diagnostics/insights-alerts-portal) based on metrics, administrative events, and other criteria.

> > *Note: Currently alerting on web app metrics is only available in the Alerts (classic) service.*

Reviewer: Use `> [!NOTE]`

Author: Done.

The [Alerts (classic) service](/azure/monitoring-and-diagnostics/monitor-quick-resource-metric-alert-portal) can be found in Azure Monitor or under the **Monitoring** section of the App Service settings.

![Alerts (classic) link](./media/monitoring/alerts.png)

## Live debugging

Azure App Service can be [debugged remotely with Visual Studio](/azure/app-service/web-sites-dotnet-troubleshoot-visual-studio#remotedebug) when logs don't provide enough information. However, remote debugging requires the app to be compiled with debug symbols. Debugging shouldn't be done in production, except as a last resort.

## Conclusion

In this section, you completed the following tasks:

* Find basic monitoring and troubleshooting data in the Azure portal
* Learn how Azure Monitor provides a deeper look at metrics across all Azure services
* Connect the web app with Application Insights for app profiling
* Turn on logging and learn where to download logs
* Stream logs in real time
* Learn where to set up alerts
* Learn about remote debugging Azure App Service web apps.

## Additional reading

> * <xref:test/troubleshoot-azure-iis>

Reviewer: ```suggestion
* [Troubleshooting ASP.NET Core on Azure App Service and IIS](/aspnet/core/test/troubleshoot-azure-iis)
```

> * <xref:host-and-deploy/azure-iis-errors-reference>

Reviewer: ```suggestion
* [Common errors reference for Azure App Service and IIS with ASP.NET Core
](/aspnet/core/host-and-deploy/azure-iis-errors-reference)
```

The [Alerts (classic) service](/azure/monitoring-and-diagnostics/monitor-quick-resource-metric-alert-portal) can be found in Azure Monitor or under the **Monitoring** section of the App Service settings.

![Alerts (classic) link](./media/monitoring/alerts.png)

## Live debugging

Azure App Service can be [debugged remotely with Visual Studio](/azure/app-service/web-sites-dotnet-troubleshoot-visual-studio#remotedebug) when logs don't provide enough information. However, remote debugging requires the app to be compiled with debug symbols. Debugging shouldn't be done in production, except as a last resort.

## Conclusion

In this section, you completed the following tasks:

* Find basic monitoring and troubleshooting data in the Azure portal
* Learn how Azure Monitor provides a deeper look at metrics across all Azure services
* Connect the web app with Application Insights for app profiling
* Turn on logging and learn where to download logs
* Stream logs in real time
* Learn where to set up alerts
* Learn about remote debugging Azure App Service web apps.

## Additional reading

* [Troubleshooting ASP.NET Core on Azure App Service and IIS](/aspnet/core/test/troubleshoot-azure-iis)

> * [Common errors reference for Azure App Service and IIS with ASP.NET Core
> ](/aspnet/core/host-and-deploy/azure-iis-errors-reference)

Reviewer: Sorry I unintentionally split it to two lines in my suggestion.

```suggestion
* [Common errors reference for Azure App Service and IIS with ASP.NET Core](/aspnet/core/host-and-deploy/azure-iis-errors-reference)
```
